executor/join.go

```diff
@@ -218,13 +221,42 @@ func (e *HashJoinExec) fetchOuterChunks(ctx context.Context) {
 			}
 			return
 		}
+
+		if !hasWaitedForInner {
+			innerErr, jobFinished := e.wait4InnerOnce()
+			if jobFinished {
+				return
+			} else if innerErr != nil {
+				e.joinResultCh <- &hashjoinWorkerResult{
+					err: errors.Trace(err),
+				}
+				return
+			}
+			hasWaitedForInner = true
+		}
+
 		if outerResult.NumRows() == 0 {
 			return
 		}
 		outerResource.dest <- outerResult
 	}
 }
 
+func (e *HashJoinExec) wait4InnerOnce() (err error, finished bool) {
```

Comment:
`error` maybe better as last return item in go's idiom~

Reply:
Yes, but I prefer to do it the same way, which use`error` as the first return value, as our existing old code does

Reply:
The function doesn't guarantee the `once`. So wait4Inner is a more suitable name.